comment_scan/replier.rb

```diff
@@ -80,68 +82,104 @@ def random_response
     end
 
     def contains_cat(message)
-        ["cat", "kitty", "kitties", "kitten", "meow", "purr", "feline"].any? { |cat_name| message.downcase.include? cat_name }
+        ["cat", "kitty", "kitties", "kitten", "kitteh", "meow", "purr", "feline"].any? { |cat_name| message.downcase.include? cat_name }
     end
 
-    def tp(msg_id, parent_id, room_id, *args)
-        comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
-        return if comment.nil?
-
-        comment.tps ||= 0
-        comment.tps += 1
-        comment.save
+    def contains_dog(message)
+        ["dog", "pup", "woof", "bark", "bow-wow", "best friend"].any? { |dog_name| message.downcase.include? dog_name }
+    end
 
-        @chatter.say "Marked this comment as caught correctly (tp). Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps. *beep boop* My human overlords won't let me flag that, so you'll have to do it yourself.", room_id
+    def animals_on?(room_id)
+      room_id == @chatter.HQroom || Room.find_by(room_id: room_id).animals 
     end
 
-    def fp(msg_id, parent_id, room_id, *args)
+
+    def record_feedback(feedback_id, parent_id, chat_user, room_id)
         comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
-        return if comment.nil?
+        return false if comment.nil?
 
+        comment.tps ||= 0
         comment.fps ||= 0
-        comment.fps += 1
-        comment.save
-        @chatter.say "Marked this comment as caught incorrectly (fp). Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps", room_id
+        comment.rude ||= 0
+
+        existing_feedback = Feedback.find_by(comment_id: comment.id, chat_user_id: chat_user.id)
+
+        if existing_feedback.nil? #Create new feedback
+            Feedback.create(comment_id: comment.id, chat_user_id: chat_user.id, feedback_type_id: feedback_id, room_id: room_id)
+        elsif existing_feedback.feedback_type_id #If one exists, then undo it
+            comment.remove_feedback(existing_feedback.feedback_type_id)
+
+            if existing_feedback.feedback_type_id == feedback_id #They're only trying to undo
+                existing_feedback.delete
+                @chatter.say "Un-#{FeedbackTypedef.feedback_name(feedback_id)}'ed this comment. Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps."
+                comment.save
+                return
```

Comment:
It feels weird to me that you don't return a boolean here while you do return a boolean at the end. Is there are reason you're doing that?